contrib/sddpg/core.c

```diff
@@ -169,12 +170,25 @@ static HB_ERRCODE pgsqlConnect( SQLDDCONNECTION * pConnection, PHB_ITEM pItem )
    PGconn *       pConn;
    ConnStatusType status;
    const char *   pszHost;
+   void **        hbConn;
+   PHB_ITEM       pSecond;
+   int existingConnection = 0;
 
-   pszHost = hb_arrayGetCPtr( pItem, 2 );
-   if( pszHost && ( strncmp( pszHost, "postgresql://", 13 ) == 0 || strchr( pszHost, '=' ) ) )
-      pConn = PQconnectdb( pszHost );
+   pSecond = hb_itemArrayGet( pItem, 2 );
+   if( HB_IS_POINTER( pSecond ) )
```

Comment:
> This will need the currently private pointer handling API in hbpgsql to be published, 
meaning moving it to a separate .c source, maybe renamings, and adding a public .h file.
Then making that contrib a dependency of this one and use its API to verify and retrieve the 
pointer.

Sorry, I have no time to work on this.